drivers/serial/uart_infineon_cat1.c

```diff
@@ -0,0 +1,506 @@
+/* Copyright 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief UART driver for Infineon CAT1 MCU family.
+ *
+ * Note:
+ * - Uart ASYNC functionality is not implemented in current
+ *   version of Uart CAT1 driver.
+ */
+
+#define DT_DRV_COMPAT infineon_mtbhal_uart
+
+#include <device.h>
+#include <drivers/uart.h>
+#include <drivers/pinctrl.h>
+#include "cyhal_uart.h"
+
+/* Default UART interrupt priority */
+#define UART_CAT1_INTERRUPT_PRIORITY      (5u)
+
+/* Map UART pins in Pinctrl array */
+#define UART_TX_PIN   (CAT1_PIN_SIGNAL_SCB_UART_TX)     /* TX pin */
+#define UART_RX_PIN   (CAT1_PIN_SIGNAL_SCB_UART_RX)     /* RX pin */
+#define UART_RTS_PIN  (CAT1_PIN_SIGNAL_SCB_UART_RTS)    /* RTS pin */
+#define UART_CTS_PIN  (CAT1_PIN_SIGNAL_SCB_UART_CTS)    /* CTS pin */
+
+/* Data structure */
+struct uart_cat1_data {
+	cyhal_uart_t obj;                          /* UART CYHAL object */
+	struct uart_config cfg;
+
+	#if CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;           /* Interrupt Callback */
+	void *irq_cb_data;                              /* Interrupt Callback Arg */
+	#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+	#ifdef CONFIG_UART_ASYNC_API
+	uart_callback_t async_cb;
+	void *async_cb_data;
+	void *async_rx_next_buf;
+	size_t async_rx_next_buf_len;
+	bool async_rx_enabled;
+	#endif /* CONFIG_UART_ASYNC_API */
+};
+
+/* Device config structure */
+struct uart_cat1_config {
+	const struct pinctrl_dev_config *pcfg;
+	struct uart_config z_cfg;
```

Comment:
Please don't use `z_` prefix in driver files.

Reply:
updated